client/src/services/UserService.js

```diff
@@ -168,6 +168,35 @@ export class UserService {
 	setDefaultWallet = async (index) => {
 		this.currentWalletIndex = parseInt(index, 10);
 	}
+
+	generateWallets = async (total) => {
+		const user = this.instance;
+		const existingWallets = user.getHDWallet()._derivedWallets || [];
```

Comment:
I'm noticing this line:
`const wallets = user.getHDWallet()._derivedWallets || [];` is being re-use multiple times in User Service.
We can consider split this into a separate method and call it from there.